packages/sample-controllers/src/sample-gas-prices-service/sample-gas-prices-service.ts

```diff
@@ -61,12 +170,19 @@ export class SampleGasPricesService {
    * @returns The gas prices for the given chain.
    */
   async fetchGasPrices(chainId: Hex) {
-    const response = await this.#fetch(
-      `https://example.com/gas-prices/${chainId}.json`,
-    );
-    // Type assertion: We have to assume the shape of the response data.
-    const gasPricesResponse =
-      (await response.json()) as unknown as GasPricesResponse;
-    return gasPricesResponse.data;
+    return await this.#policy.execute(async () => {
+      const response = await this.#fetch(
+        `https://example.com/gas-prices/${chainId}.json`,
+      );
+      if (response.ok) {
+        // Type assertion: We have to assume the shape of the response data.
```

Comment:
Nit (pre-existing issue): We can validate, we definitely don't need to assume the shape. Assuming the shape of the response without validation should be strongly discouraged. We've had plenty of examples even with internal APIs of them not returning the expected shape (especially with error responses, which sometimes can be returned even by HTTP 200 responses)

Reply:
You're right, but we haven't been validating responses elsewhere as far as I know. What is the pattern that we want to promote here, should we use Superstruct to do so or do it manually? And what should happen if the validation fails, should we throw an error?

Reply:
We don't have any prescriptive guidance right now on how to best validate, so that would be up to the team or contributor.

Either throwing an error for invalid inputs, or stripping the data down to the valid parts/normalizing it, may be appropriate depending on the circumstance. Personally I'd default to throwing an error if I wasn't sure how to handle something.